.github/scripts/test_check_action_pins.py

```
@@ -0,0 +1,504 @@
+            text=True,
+            timeout=60,
+            cwd=self.root,
+        )
+        findings = [
+            line.strip() for line in result.stderr.splitlines() if line.startswith("  ")
+        ]
+        return result.returncode, findings, result.stderr
+
+    def scan_targets(self) -> list[str]:
+        """The guard's own discovery list, as repo-relative strings."""
+        listing = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)